astroquery/esa/utils/utils.py

```diff
@@ -7,11 +7,437 @@
+        """
+
+        if user is None:
+            user = input("Username:")
+        if password is None:
+            password = getpass.getpass("Password:")
+
+        if user and password:
+            args = {
+                "username": str(user),
+                "password": str(password)}
+            header = {
+                "Content-type": "application/x-www-form-urlencoded",
+                "Accept": "text/plain"
+            }
+            response = self.post(url=login_url, data=args, headers=header)
+            try:
+                response.raise_for_status()
+                log.info('User has been logged successfully.')
+            except Exception as e:
+                log.error('Logging error: {}'.format(e))
+                raise e
+
+    def logout(self, logout_url):
+        """
+        Performs a logout.
+        TAP+ only
+        User and password shall be used
+
+        Parameters
+        ----------
+        logout_url: str, mandatory
+            URL to execute the logout request
+        """
+        header = {
+            "Content-type": "application/x-www-form-urlencoded",
+            "Accept": "text/plain"
+        }
+        response = self.post(url=logout_url, headers=header)
+        try:
+            response.raise_for_status()
+            log.info('Logout executed successfully.')
+        except Exception as e:
+            log.error('Logout error: {}'.format(e))
+            raise e
+
+    def _send_login_request(self, username: str, password: str) -> Response:
+        """
+        Send a POST request to the login URL with username and password.
+
+        Parameters:
+            username (str): The username for authentication.
+            password (str): The password for authentication.
+
+        Returns:
+            Response: The response object from the login request.
+        """
+        import requests  # Use requests for handling HTTP interactions
+
+        # Login payload
+        payload = {"username": username, "password": password}
+
+        # Perform the POST request
+        response = requests.post(self.login_url, data=payload)
+
+        # Raise an error if the request failed
+        response.raise_for_status()
+        return response
+
+    def _request(self, method, url, *args, **kwargs):
+        """
+        Intercept the request method and add TAPCLIENT=ASTROQUERY
+
+        Parameters
+        ----------
+        method: str, mandatory
+            method to be executed
+        url: str, mandatory
+            url for the request
+
+        Returns
+        -------
+        The request with the modified url
+        """
+
+        # Add the custom query parameter to the URL
+        if '?' in url:
+            url += "&TAPCLIENT=ASTROQUERY&format=votable_plain"
+        else:
+            url += "?TAPCLIENT=ASTROQUERY&format=votable_plain"
+        return super()._request(method, url, **kwargs)
+
+
+def get_coord_input(value, msg):
```

Comment:
most of these are generic enough, so may need to be cleaned up/consolidated. E.g. why not just use `commons.parse_coordinates` in the integral module?

Reply:
Done.